src/api/web3/index.ts

```diff
@@ -1,13 +1,23 @@
 import Web3 from 'web3'
+import { getNetworkFromId } from '@gnosis.pm/dex-js'
 
-import { ETH_NODE_URL } from 'const'
+import { Network } from 'types'
+
+import { ETH_NODE_URL, INFURA_ID } from 'const'
 
 // TODO connect to mainnet if we need AUTOCONNECT at all
 export const getDefaultProvider = (): string | null => (process.env.NODE_ENV === 'test' ? null : ETH_NODE_URL)
 
-export function createWeb3Api(): Web3 {
+const web3cache = {}
```

Comment:
Probably you would want to be more specific about the type, right?
Like using `Record<string, Web3>`

Reply:
This is only needed if we go for approach 1 (1 instance per network).
We can do that anyway, but I'm more in favor of going for approach 2